apps/condo/domains/acquiring/tasks/recurrent-payments-context/recurrent-payment-context-processing.js

```diff
@@ -0,0 +1,93 @@
+const dayjs = require('dayjs')
+const { get } = require('lodash')
+const { v4: uuid } = require('uuid')
+
+const { getLogger } = require('@open-condo/keystone/logging')
+const { getSchemaCtx } = require('@open-condo/keystone/schema')
+const { createCronTask } = require('@open-condo/keystone/tasks')
+
+const {
+    dvAndSender,
+    paginationConfiguration,
+} = require('@condo/domains/acquiring/tasks/utils/constants')
+const {
+    getAllReadyToPayRecurrentPaymentContexts,
+    getReceiptsForServiceConsumer,
+} = require('@condo/domains/acquiring/tasks/utils/queries')
+const {
+    RecurrentPayment,
+} = require('@condo/domains/acquiring/utils/serverSchema')
+const { processArrayOf } = require('@condo/domains/common/utils/parallel')
+
+const logger = getLogger('recurrent-payment-context-processing')
+
+async function processContext (context, date, recurrentPaymentContext) {
+    // prepare vars
+    const { serviceConsumer, billingCategory } = recurrentPaymentContext
+    const previousMonthDate = date.startOf('month').subtract(1, 'days')
+
+    // get billing receipts
+    const billingReceipts = await getReceiptsForServiceConsumer(
+        context,
+        previousMonthDate,
+        serviceConsumer,
+        billingCategory,
+    )
+
+    // no receipts == no payment tasks
+    if (billingReceipts.length === 0) {
+        return
+    }
+
+    // create payment proceeding tasks
+    return await RecurrentPayment.create(context, {
+        ...dvAndSender,
+        tryCount: 0,
+        state: {},
+        billingReceipts: billingReceipts.map(receipt => ({ id: receipt.id })),
+        recurrentPaymentContext: { connect: { id: recurrentPaymentContext.id } },
+    })
+}
+
+async function process () {
+    const jobId = uuid()
```

Comment:
you can use `cont task = this; const taskId = task.id`

Reply:
You can look at: https://github.com/open-condo-software/condo/blob/master/packages/keystone/tasks.js#L232-L238 to understan how really worker works

Reply:
I'll keep it as task.id || uuid() - since we can run scripts manually